diff -ur asterisk-1.2.7.1-o/apps/Makefile asterisk-1.2.7.1/apps/Makefile
--- asterisk-1.2.7.1-o/apps/Makefile	2006-02-08 19:31:21.000000000 -0700
+++ asterisk-1.2.7.1/apps/Makefile	2006-05-05 18:20:40.000000000 -0600
@@ -54,6 +54,10 @@
 APPS+=app_osplookup.so
 endif
 
+ifneq ($(wildcard $(CROSS_COMPILE_TARGET)/usr/local/include/spandsp.h $(CROSS_COMPILE_TARGET)/usr/include/spandsp.h),)
+APPS+=app_rxfax.so app_txfax.so
+endif
+
 ifeq ($(findstring BSD,${OSARCH}),BSD)
 CFLAGS+=-I$(CROSS_COMPILE_TARGET)/usr/local/include -L$(CROSS_COMPILE_TARGET)/usr/local/lib
 endif
@@ -102,6 +106,12 @@
 app_curl.so: app_curl.o
 	$(CC) $(SOLINK) -o $@ ${CYGSOLINK} $< ${CYGSOLIB} $(CURLLIBS)
 
+app_rxfax.so : app_rxfax.o
+	$(CC) $(SOLINK) -o $@ $< -lspandsp -ltiff
+ 
+app_txfax.so : app_txfax.o
+	$(CC) $(SOLINK) -o $@ $< -lspandsp -ltiff
+ 
 app_sql_postgres.o: app_sql_postgres.c
 	$(CC) -pipe -I$(CROSS_COMPILE_TARGET)/usr/local/pgsql/include -I$(CROSS_COMPILE_TARGET)/usr/include/postgresql $(CFLAGS) -c -o app_sql_postgres.o app_sql_postgres.c
 
Only in asterisk-1.2.7.1/apps: Makefile~
Only in asterisk-1.2.7.1/apps: Makefile.orig
