diff -urN asterisk-1.4.0-beta3-o-o/acinclude.m4 asterisk-1.4.0-beta3/acinclude.m4
--- asterisk-1.4.0-beta3-o-o/acinclude.m4	2006-09-19 11:07:22.000000000 -0600
+++ asterisk-1.4.0-beta3/acinclude.m4	2006-11-06 12:44:39.000000000 -0700
@@ -146,7 +146,7 @@
       else
         AC_CHECK_FILE(/usr/include/ptlib.h, HAS_PWLIB=1, )
         if test "${HAS_PWLIB:-unset}" != "unset" ; then
-          AC_PATH_PROG(PTLIB_CONFIG, ptlib-config, , /usr/share/pwlib/make)
+          AC_PATH_PROG(PTLIB_CONFIG, ptlib-config, , /usr/bin)
           PWLIB_INCDIR="/usr/include"
           PWLIB_LIBDIR="/usr/lib"
         fi
