--- asterisk-1.2.23/channels/Makefile~	2007-07-31 21:41:40.362172509 +0200
+++ asterisk-1.2.23/channels/Makefile	2007-07-31 21:44:00.379216123 +0200
@@ -120,7 +120,7 @@ ifneq ($(wildcard alsa-monitor.h),)
 endif
 
 ifndef WITHOUT_ZAPTEL
-ifneq ($(wildcard $(CROSS_COMPILE_TARGET)/usr/include/linux/zaptel.h)$(wildcard $(CROSS_COMPILE_TARGET)/usr/local/include/zaptel.h)$(wildcard $(CROSS_COMPILE_TARGET)/usr/pkg/include/zaptel.h),)
+ifneq ($(wildcard $(CROSS_COMPILE_TARGET)/usr/include/zaptel/zaptel.h)$(wildcard $(CROSS_COMPILE_TARGET)/usr/local/include/zaptel.h)$(wildcard $(CROSS_COMPILE_TARGET)/usr/pkg/include/zaptel.h),)
   ifeq (${OSARCH},NetBSD)
     SOLINK+=-L$(CROSS_COMPILE_TARGET)/usr/pkg/lib
   endif
--- asterisk-1.2.23/channels/chan_iax2.c~	2007-07-24 18:32:04.000000000 +0200
+++ asterisk-1.2.23/channels/chan_iax2.c	2007-07-31 21:50:35.148751179 +0200
@@ -50,7 +50,7 @@
 #ifdef IAX_TRUNKING
 #include <sys/ioctl.h>
 #ifdef __linux__
-#include <linux/zaptel.h>
+#include <zaptel/zaptel.h>
 #else
 #include <zaptel.h>
 #endif /* __linux__ */
--- asterisk-1.2.23/channels/chan_zap.c~	2007-07-31 21:54:28.509379501 +0200
+++ asterisk-1.2.23/channels/chan_zap.c	2007-07-31 21:54:52.237748704 +0200
@@ -49,12 +49,12 @@
 #include <unistd.h>
 #include <sys/ioctl.h>
 #ifdef __linux__
-#include <linux/zaptel.h>
+#include <zaptel/zaptel.h>
 #else
 #include <zaptel.h>
 #endif /* __linux__ */
 #include <math.h>
-#include <tonezone.h>
+#include <zaptel/tonezone.h>
 #include <ctype.h>
 #ifdef ZAPATA_PRI
 #include <libpri.h>
