diff -urN asterisk-1.0.7.org/channels/h323/Makefile asterisk-1.0.7/channels/h323/Makefile
--- asterisk-1.0.7.org/channels/h323/Makefile	2004-11-15 03:35:35.000000000 +0100
+++ asterisk-1.0.7/channels/h323/Makefile	2005-03-20 20:19:02.386545728 +0100
@@ -77,7 +77,7 @@
 	ar cr libchanh323.a ast_h323.o
 
 ast_h323.o:	ast_h323.cpp
-	$(CXX) -g -c -fno-rtti -o $@ $(CFLAGS) $<
+	$(CXX) -g -c -o $@ $(CFLAGS) $<
 
 ifneq ($(wildcard .depend),)
 include .depend
