diff -urN asterisk-1.0.7.org/channels/h323/ast_h323.cpp asterisk-1.0.7/channels/h323/ast_h323.cpp
--- asterisk-1.0.7.org/channels/h323/ast_h323.cpp	2004-11-05 00:14:26.000000000 +0100
+++ asterisk-1.0.7/channels/h323/ast_h323.cpp	2005-03-20 20:27:59.908829920 +0100
@@ -994,6 +994,7 @@
 	} else {
 		endPoint->SetSendUserInputMode(H323Connection::SendUserInputAsInlineRFC2833);
 	}
+#if 0
 	if (cap & AST_FORMAT_SPEEX) {
 		/* Not real sure if Asterisk acutally supports all
 		   of the various different bit rates so add them 
@@ -1005,6 +1006,7 @@
 		endPoint->SetCapability(0, 0, new SpeexNarrow5AudioCapability());
 		endPoint->SetCapability(0, 0, new SpeexNarrow6AudioCapability());
 	}
+#endif
 
 	if (cap & AST_FORMAT_G729A) {
 		AST_G729ACapability *g729aCap;
@@ -1018,11 +1020,13 @@
 		endPoint->SetCapability(0, 0, g7231Cap = new H323_G7231Capability);
 	} 
 
+#if 0
 	if (cap & AST_FORMAT_GSM) {
 		H323_GSM0610Capability *gsmCap;
 	    	endPoint->SetCapability(0, 0, gsmCap = new H323_GSM0610Capability);
 	    	gsmCap->SetTxFramesInPacket(gsmFrames);
 	} 
+#endif
 
 	if (cap & AST_FORMAT_ULAW) {
 		H323_G711Capability *g711uCap;
