--- ./channels/h323/Makefile.org	Tue May 13 09:53:16 2003
+++ ./channels/h323/Makefile	Tue May 13 09:55:26 2003
@@ -61,13 +62,13 @@
 	g++ -g -c -o $@ $(CFLAGS) $<
 
 chan_h323.so:	chan_h323.o ast_h323.o
-	g++  -shared -Xlinker -x -o chan_h323.so chan_h323.o ast_h323.o -L$(PWLIBDIR)/lib  -lpt_linux_x86_r -L$(OPENH323DIR)/lib -lh323_linux_x86_r -L/usr/lib -lpthread -ldl -lcrypto -lssl -lexpat
+	g++  -shared -Xlinker -x -o chan_h323.so chan_h323.o ast_h323.o -L$(PWLIBDIR)/lib  -lpt -L$(OPENH323DIR)/lib -lopenh323 -L/usr/lib -lpthread -ldl -lcrypto -lssl -lexpat
 
 chan_h323_d.so:	chan_h323.o ast_h323.o
-	g++  -shared -Xlinker -x -o chan_h323.so chan_h323.o ast_h323.o -L$(PWLIBDIR)/lib  -lpt_linux_x86_d -L$(OPENH323DIR)/lib -lh323_linux_x86_d -L/usr/lib -lpthread -ldl -lcrypto -lssl -lexpat
+	g++  -shared -Xlinker -x -o chan_h323.so chan_h323.o ast_h323.o -L$(PWLIBDIR)/lib  -lpt -L$(OPENH323DIR)/lib -lopenh323 -L/usr/lib -lpthread -ldl -lcrypto -lssl -lexpat
 
 chan_h323_s.so:	chan_h323.o ast_h323.o
-	g++  -shared -Xlinker -x -o chan_h323.so chan_h323.o ast_h323.o -L$(PWLIBDIR)/lib  -lpt_linux_x86_r_s -L$(OPENH323DIR)/lib -lh323_linux_x86_r_s -L/usr/lib -lpthread -ldl -lcrypto -lssl -lexpat
+	g++  -shared -Xlinker -x -o chan_h323.so chan_h323.o ast_h323.o -L$(PWLIBDIR)/lib  -lpt -L$(OPENH323DIR)/lib -lopenh323 -L/usr/lib -lpthread -ldl -lcrypto -lssl -lexpat
 clean:
 	rm -f *.o *.so core.*
 
