diff -dur asterisk-12.0.0.orig/main/asterisk.c asterisk-12.0.0/main/asterisk.c
--- asterisk-12.0.0.orig/main/asterisk.c	2014-01-07 15:18:43.000000000 +0100
+++ asterisk-12.0.0/main/asterisk.c	2014-01-07 15:18:21.000000000 +0100
@@ -223,7 +223,6 @@
 #include "asterisk/utils.h"
 #include "asterisk/file.h"
 #include "asterisk/io.h"
-#include "editline/histedit.h"
 #include "asterisk/config.h"
 #include "asterisk/ast_version.h"
 #include "asterisk/linkedlists.h"
