--- asterisk-1.2.0/codecs/gsm/Makefile.orig	2005-11-21 13:34:00.000000000 +0100
+++ asterisk-1.2.0/codecs/gsm/Makefile	2005-11-21 13:40:02.000000000 +0100
@@ -235,6 +235,7 @@
 ifneq ($(shell uname -m),alpha)
 ifneq ($(shell uname -m),armv4l)
 ifneq ($(shell uname -m),sparc64)
+ifneq ($(shell uname -m),sparc)
 ifneq (${PROC},arm)
 GSM_SOURCES+= $(SRC)/k6opt.s
 endif
@@ -245,6 +246,7 @@
 endif
 endif
 endif
+endif
 
 TOAST_SOURCES = $(SRC)/toast.c 		\
 		$(SRC)/toast_lin.c	\
@@ -296,6 +298,7 @@
 ifneq ($(shell uname -m), ppc64)
 ifneq ($(shell uname -m), alpha)
 ifneq ($(shell uname -m), sparc64)
+ifneq ($(shell uname -m), sparc)
 ifneq ($(shell uname -m), armv4l)
 GSM_OBJECTS+= $(SRC)/k6opt.o
 endif
@@ -305,6 +308,7 @@
 endif
 endif
 endif
+endif
 
 TOAST_OBJECTS =	$(SRC)/toast.o 		\
 		$(SRC)/toast_lin.o	\
