diff -urN asterisk/channels/h323/Makefile asterisk.the.good/channels/h323/Makefile
--- asterisk/channels/h323/Makefile	2003-08-11 18:39:52.000000000 +0200
+++ asterisk.the.good/channels/h323/Makefile	2003-08-11 18:33:41.000000000 +0200
@@ -24,7 +24,7 @@
 CFLAGS += -DP_HAS_SEMAPHORES -DP_SSL -DP_PTHREADS
 CFLAGS += -DPHAS_TEMPLATES -DPTRACING -DP_USE_PRAGMA 
 CFLAGS += -I$(PWLIBDIR)/include/ptlib/unix -I$(PWLIBDIR)/include 
-CFLAGS += -I$(OPENH323DIR)/include -I/usr/include/openh323
+CFLAGS += -I$(OPENH323DIR)/include -I/usr/include/openh323 -I../../include
 CFLAGS += -Wno-missing-prototypes -Wno-missing-declarations
 
 
