diff -urbB asterisk-1.0.9.org/editline/makelist asterisk-1.0.9/editline/makelist
--- asterisk-1.0.9.org/editline/makelist	2002-11-27 06:04:06.000000000 +0100
+++ asterisk-1.0.9/editline/makelist	2005-10-09 00:06:41.642346000 +0200
@@ -39,7 +39,7 @@
 
 # makelist.sh: Automatically generate header files...
 
-AWK=/usr/bin/awk
+AWK=awk
 USAGE="Usage: $0 -h|-e|-fc|-fh|-bc|-bh|-m <filenames>"
 
 if [ "x$1" = "x" ]
