diff -urbB asterisk-1.0.9.org/channels/h323/Makefile asterisk-1.0.9/channels/h323/Makefile
--- asterisk-1.0.9.org/channels/h323/Makefile	2004-11-15 03:35:35.000000000 +0100
+++ asterisk-1.0.9/channels/h323/Makefile	2005-10-09 11:49:52.396650250 +0200
@@ -20,11 +20,6 @@
 # This needs to be updated to deal with more than just little endian machines
 #
 OSARCH=$(shell uname -s)
-ifneq (${OSARCH},FreeBSD)
-ifneq (${OSARCH},NetBSD)
-CFLAGS += -march=$(shell uname -m)
-endif
-endif
 CFLAGS += -DPBYTE_ORDER=PLITTLE_ENDIAN
 
 ifeq (${OSARCH},Linux)
Tylko w asterisk-1.0.9/channels/h323: Makefile~
diff -urbB asterisk-1.0.9.org/codecs/gsm/Makefile asterisk-1.0.9/codecs/gsm/Makefile
--- asterisk-1.0.9.org/codecs/gsm/Makefile	2005-06-21 16:27:28.000000000 +0200
+++ asterisk-1.0.9/codecs/gsm/Makefile	2005-10-09 11:50:18.350272250 +0200
@@ -37,34 +37,11 @@
 ######### ppro's, etc, as well as the AMD K6 and K7.  The compile will
 ######### probably require gcc. 
 
-ifneq (${OSARCH},Darwin)
-ifneq (${PROC},x86_64)
-ifneq (${PROC},ultrasparc)
-ifneq ($(shell uname -m),ppc)
-ifneq ($(shell uname -m),alpha)
-ifneq ($(shell uname -m),armv4l)
-ifneq (${PROC},sparc64)
-ifneq (${PROC},ppc)
-ifneq (${PROC},ppc64)
-OPTIMIZE+=-march=$(PROC)
-endif
-endif
-endif
-endif
-endif
-endif
-endif
-endif
-endif
-
 #The problem with sparc is the best stuff is in newer versions of gcc (post 3.0) only.
 #This works for even old (2.96) versions of gcc and provides a small boost either way.
 #A ultrasparc cpu is really v9 but the stock debian stable 3.0 gcc doesn't support it.
 #So we go lowest common available by gcc and go a step down, still a step up from
 #the default as we now have a better instruction set to work with. - Belgarath
-ifeq (${PROC},ultrasparc)
-OPTIMIZE+=-mcpu=v8 -mtune=$(PROC) -O3 -fomit-frame-pointer
-endif
 
 PG =
 #PG = -g -pg
Tylko w asterisk-1.0.9/codecs/gsm: Makefile~
diff -urbB asterisk-1.0.9.org/codecs/lpc10/Makefile asterisk-1.0.9/codecs/lpc10/Makefile
--- asterisk-1.0.9.org/codecs/lpc10/Makefile	2004-08-31 18:33:00.000000000 +0200
+++ asterisk-1.0.9/codecs/lpc10/Makefile	2005-10-09 11:50:32.795175000 +0200
@@ -25,28 +25,6 @@
 CFLAGS += $(OPTIMIZE) -I$(LIB_TARGET_DIR) $(WARNINGS) -fPIC
 #CFLAGS+= $(shell if uname -m | grep -q 86; then echo "-mpentium" ; fi)
 
-#fix for PPC processors and ALPHA, And UltraSparc too
-ifneq ($(OSARCH),Darwin)
-ifneq ($(findstring BSD,${OSARCH}),BSD)
-ifneq ($(PROC),ppc)
-ifneq ($(PROC),x86_64)
-ifneq ($(PROC),alpha)
-#The problem with sparc is the best stuff is in newer versions of gcc (post 3.0) only.
-#This works for even old (2.96) versions of gcc and provides a small boost either way.
-#A ultrasparc cpu is really v9 but the stock debian stable 3.0 gcc doesn.t support it.
-#So we go lowest common available by gcc and go a step down, still a step up from
-#the default as we now have a better instruction set to work with. - Belgarath
-ifeq ($(PROC),ultrasparc)
-        CFLAGS+= -mtune=$(PROC) -mcpu=v8 -O3 -fomit-frame-pointer
-else
-        CFLAGS+= -march=$(PROC)
-endif
-endif
-endif
-endif
-endif
-endif
-
 LIB = $(LIB_TARGET_DIR)/liblpc10.a
 
 .PHONY: all clean
Tylko w asterisk-1.0.9/codecs/lpc10: Makefile~
diff -urbB asterisk-1.0.9.org/Makefile asterisk-1.0.9/Makefile
--- asterisk-1.0.9.org/Makefile	2005-04-26 16:30:23.000000000 +0200
+++ asterisk-1.0.9/Makefile	2005-10-09 11:49:39.263829500 +0200
@@ -130,10 +130,6 @@
 CFLAGS=-pipe  -Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations $(DEBUG) $(INCLUDE) -D_REENTRANT -D_GNU_SOURCE #-DMAKE_VALGRIND_HAPPY
 CFLAGS+=$(OPTIMIZE)
 
-ifneq ($(PROC),ultrasparc)
-CFLAGS+=$(shell if $(CC) -march=$(PROC) -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then echo "-march=$(PROC)"; fi)
-endif
-
 CFLAGS+=$(shell if uname -m | grep -q ppc; then echo "-fsigned-char"; fi)
 CFLAGS+=$(shell if [ -f /usr/include/osp/osp.h ]; then echo "-DOSP_SUPPORT -I/usr/include/osp" ; fi)
 
