--- asterisk-13.4.0/codecs/codec_ilbc.c.orig	2015-06-04 15:20:51.000000000 +0200
+++ asterisk-13.4.0/codecs/codec_ilbc.c	2015-08-01 21:55:41.236727967 +0200
@@ -40,8 +40,8 @@
 
 #ifdef ILBC_WEBRTC
 #include <ilbc.h>
-typedef WebRtc_UWord16 ilbc_bytes;
-typedef WebRtc_Word16  ilbc_block;
+typedef uint16_t ilbc_bytes;
+typedef int16_t  ilbc_block;
 #define BUF_TYPE i16
 #else
 #include "ilbc/iLBC_encode.h"
