From 346965149827844aa5a5f06ab155787e54a70e30 Mon Sep 17 00:00:00 2001
From: Jeffrey C. Ollie <jeff@ocjtech.us>
Date: Mon, 8 Mar 2010 12:38:56 -0600
Subject: [PATCH 2/2] Use the library function for loading command history rather than
 implementing our own.

--- asterisk-1.6.1.12/main/asterisk.c~	2010-09-16 17:31:01.000000000 +0300
+++ asterisk-1.6.1.12/main/asterisk.c	2010-09-16 17:32:29.551700797 +0300
@@ -2431,30 +2431,12 @@
 
 static int ast_el_read_history(char *filename)
 {
-	char buf[256];
-	FILE *f;
-	int ret = -1;
+	HistEvent ev;
 
 	if (el_hist == NULL || el == NULL)
 		ast_el_initialize();
 
-	if ((f = fopen(filename, "r")) == NULL)
-		return ret;
-
-	while (!feof(f)) {
-		if (!fgets(buf, sizeof(buf), f)) {
-			continue;
-		}
-		if (!strcmp(buf, "_HiStOrY_V2_\n"))
-			continue;
-		if (ast_all_zeros(buf))
-			continue;
-		if ((ret = ast_el_add_history(buf)) == -1)
-			break;
-	}
-	fclose(f);
-
-	return ret;
+	return (history(el_hist, &ev, H_LOAD, filename));
 }
 
 static void ast_remotecontrol(char *data)
