--- asterisk-1.6.1.12/main/asterisk.c~	2009-12-30 23:34:17.000000000 +0200
+++ asterisk-1.6.1.12/main/asterisk.c	2009-12-30 23:34:19.095481746 +0200
@@ -110,7 +110,7 @@
 #include "asterisk/utils.h"
 #include "asterisk/file.h"
 #include "asterisk/io.h"
-#include "editline/histedit.h"
+#include <histedit.h>
 #include "asterisk/config.h"
 #include "asterisk/ast_version.h"
 #include "asterisk/linkedlists.h"
@@ -2291,7 +2291,7 @@
 			int mlen = 0, maxmbuf = 2048;
 			/* Start with a 2048 byte buffer */			
 			if (!(mbuf = ast_malloc(maxmbuf))) {
-				lf->cursor[0] = savechr;
+//				lf->cursor[0] = savechr; // this is readonly variable
 				return (char *)(CC_ERROR);
 			}
 			snprintf(buf, sizeof(buf), "_COMMAND MATCHESARRAY \"%s\" \"%s\"", lf->buffer, ptr); 
@@ -2303,7 +2303,7 @@
 					/* Every step increment buffer 1024 bytes */
 					maxmbuf += 1024;					
 					if (!(mbuf = ast_realloc(mbuf, maxmbuf))) {
-						lf->cursor[0] = savechr;
+//						lf->cursor[0] = savechr; // this is readonly variable
 						return (char *)(CC_ERROR);
 					}
 				}
@@ -2365,7 +2365,7 @@
 		ast_free(matches);
 	}
 
-	lf->cursor[0] = savechr;
+//	lf->cursor[0] = savechr; // this is readonly variable
 
 	return (char *)(long)retval;
 }
