diff -urN asterisk-16.15.0.org/build_tools/menuselect-deps.in asterisk-16.15.0/build_tools/menuselect-deps.in
--- asterisk-16.15.0.org/build_tools/menuselect-deps.in	2020-11-19 13:34:40.000000000 +0100
+++ asterisk-16.15.0/build_tools/menuselect-deps.in	2020-11-27 12:46:51.858456198 +0100
@@ -14,6 +14,7 @@
 GENERIC_ODBC=@PBX_GENERIC_ODBC@
 GMIME=@PBX_GMIME@
 GNU_LD=@GNU_LD@
+LPC10=@PBX_LPC10@
 GSM=@PBX_GSM@
 ILBC=@PBX_ILBC@
 GTK2=@PBX_GTK2@
diff -urN asterisk-16.15.0.org/codecs/codec_lpc10.c asterisk-16.15.0/codecs/codec_lpc10.c
--- asterisk-16.15.0.org/codecs/codec_lpc10.c	2020-11-19 13:34:40.000000000 +0100
+++ asterisk-16.15.0/codecs/codec_lpc10.c	2020-11-27 12:46:51.858456198 +0100
@@ -26,6 +26,9 @@
  *
  * \ingroup codecs
  */
+/*** MODULEINFO
+	<depend>lpc10</depend>
+ ***/
 
 /*** MODULEINFO
 	<support_level>core</support_level>
@@ -39,7 +42,7 @@
 #include "asterisk/utils.h"
 #include "asterisk/linkedlists.h"
 
-#include "lpc10/lpc10.h"
+#include <lpc10.h>
 
 /* Sample frame data */
 #include "asterisk/slin.h"
diff -urN asterisk-16.15.0.org/codecs/Makefile asterisk-16.15.0/codecs/Makefile
--- asterisk-16.15.0.org/codecs/Makefile	2020-11-19 13:34:40.000000000 +0100
+++ asterisk-16.15.0/codecs/Makefile	2020-11-27 12:46:51.858456198 +0100
@@ -27,8 +27,7 @@
 
 SUB_DIRS := \
 	$(SUB_GSM) \
-	$(SUB_ILBC) \
-	$(SUB_LPC10) \
+	$(SUB_ILBC)
 
 all: _all
 
@@ -57,7 +56,9 @@
 .PHONY: $(SUB_DIRS)
 
 
+ifneq ($(LPC10_INTERNAL),no)
 codec_lpc10.so: $(LIBLPC10)
+endif
 
 # Don't run the implicit rules for this target.
 $(LIBLPC10): $(SUB_LPC10) ;
diff -urN asterisk-16.15.0.org/configure.ac asterisk-16.15.0/configure.ac
--- asterisk-16.15.0.org/configure.ac	2020-11-27 12:46:42.094818363 +0100
+++ asterisk-16.15.0/configure.ac	2020-11-27 12:46:51.858456198 +0100
@@ -488,6 +488,7 @@
 AST_EXT_LIB_SETUP([GSM], [External GSM], [gsm], [, use 'internal' GSM otherwise])
 AST_EXT_LIB_SETUP([ILBC], [System iLBC], [ilbc], [, use 'internal' iLBC otherwise])
 AST_EXT_LIB_SETUP([GTK2], [gtk2], [gtk2])
+AST_EXT_LIB_SETUP([LPC10], [External lpc10 library], [lpc10], [, use 'internal' lpc10 otherwise])
 AST_EXT_LIB_SETUP([GMIME], [GMime], [gmime])
 AST_EXT_LIB_SETUP([HOARD], [Hoard Memory Allocator], [hoard])
 AST_EXT_LIB_SETUP([ICAL], [iCal], [ical])
@@ -1661,6 +1662,72 @@
    fi
 fi
 
+LPC10_INTERNAL="yes"
+AC_SUBST(LPC10_INTERNAL)
+LPC10_SYSTEM="yes"
+if test "${USE_LPC10}" != "no"; then
+   case "$host_cpu" in
+   x86_64|s390x)
+       _lib=lib64
+       ;;
+   *)
+       _lib=lib
+       ;;
+   esac
+
+   if test "${LPC10_DIR}" = "internal"; then
+      LPC10_SYSTEM="no"
+   elif test "${LPC10_DIR}" != ""; then
+      LPC10_INTERNAL="no"
+   fi
+   if test "${LPC10_SYSTEM}" = "yes"; then
+      lpc10libdir=""
+      if test "x${LPC10_DIR}" != "x"; then
+         if test -d ${LPC10_DIR}/${_lib}; then
+            lpc10libdir="-L${LPC10_DIR}/${_lib}"
+         else
+            lpc10libdir="-L${LPC10_DIR}"
+         fi
+      fi
+      AC_CHECK_LIB([lpc10], [lpc10_encode], AC_DEFINE_UNQUOTED([HAVE_LPC10], 1,
+      [Define to indicate the GSM library]), [], ${gsmlibdir})
+      if test "${ac_cv_lib_lpc10_lpc10_encode}" = "yes"; then
+         if test "x${LPC10_DIR}" != "x" ; then
+            AC_CHECK_HEADER([${LPC10_DIR}/include/lpc10.h], [LPC10_HEADER_FOUND=1], [LPC10_HEADER_FOUND=0])
+         else
+            AC_CHECK_HEADER([lpc10.h], [LPC10_HEADER_FOUND=1], [LPC10_HEADER_FOUND=0])
+         fi
+         if test "${LPC10_HEADER_FOUND}" = "0" ; then
+		   if test "x${LPC10_MANDATORY}" = "xyes" ; then
+			  AC_MSG_NOTICE([***])
+			  AC_MSG_NOTICE([*** It appears that you do not have the lpc10 development package installed.])
+			  AC_MSG_NOTICE([*** Please install it to include ${LPC10_DESCRIP} support, or re-run configure])
+			  AC_MSG_NOTICE([*** without explicitly specifying --with-${LPC10_OPTION}])
+			  exit 1
+		   fi
+         fi
+         LPC10_OK=0
+         if test "${LPC10_HEADER_FOUND}" = "1" ; then
+            AC_DEFINE_UNQUOTED([HAVE_LPC10_HEADER], 1, [Define to indicate that lpc10.h has no prefix for its location])
+            LPC10_OK=1
+         fi
+         if test "${LPC10_OK}" = "1" ; then
+            LPC10_LIB="-llpc10"
+            if test "x${LPC10_DIR}" != "x"; then
+               LPC10_LIB="${lpc10libdir} ${LPC10_LIB}"
+               LPC10_INCLUDE="-I${LPC10_DIR}/include"
+            fi
+            PBX_LPC10=1
+            LPC10_INTERNAL="no"
+         fi
+      fi
+   fi
+   if test "${LPC10_INTERNAL}" = "yes"; then
+      PBX_LPC10=1
+      AC_DEFINE_UNQUOTED([HAVE_LPC10_HEADER], 1, [Define to indicate that lpc10.h has no prefix for its location])
+   fi
+fi
+
 AST_EXT_LIB_CHECK([ICONV], [iconv], [iconv_open], [iconv.h])
 # GNU libiconv #define's iconv_open to libiconv_open, so we need to search for that symbol
 AST_EXT_LIB_CHECK([ICONV], [iconv], [libiconv_open], [iconv.h])
diff -urN asterisk-16.15.0.org/makeopts.in asterisk-16.15.0/makeopts.in
--- asterisk-16.15.0.org/makeopts.in	2020-11-19 13:34:40.000000000 +0100
+++ asterisk-16.15.0/makeopts.in	2020-11-27 12:46:51.858456198 +0100
@@ -160,6 +160,10 @@
 ILBC_INCLUDE=@ILBC_INCLUDE@
 ILBC_LIB=@ILBC_LIB@
 
+LPC10_INTERNAL=@LPC10_INTERNAL@
+LPC10_INCLUDE=@LPC10_INCLUDE@
+LPC10_LIB=@LPC10_LIB@
+
 GTK2_INCLUDE=@GTK2_INCLUDE@
 GTK2_LIB=@GTK2_LIB@
 
