diff -dur asterisk-12.0.0.orig/configure.ac asterisk-12.0.0/configure.ac
--- asterisk-12.0.0.orig/configure.ac	2013-12-03 18:59:41.000000000 +0100
+++ asterisk-12.0.0/configure.ac	2014-01-07 09:42:07.000000000 +0100
@@ -2113,12 +2113,12 @@
 		[${PWLIB_INCLUDE}], [${PWLIB_LIB}])
 fi
 
-AST_EXT_LIB_CHECK([LUA], [lua5.1], [luaL_newstate], [lua5.1/lua.h], [-lm])
+AST_EXT_LIB_CHECK([LUA], [lua51], [luaL_newstate], [lua51/lua.h], [-lm])
 if test "x${PBX_LUA}" = "x1" ; then
 	if test x"${LUA_DIR}" = x; then
-		LUA_INCLUDE="${LUA_INCLUDE} -I/usr/include/lua5.1"
+		LUA_INCLUDE="${LUA_INCLUDE} -I/usr/include/lua51"
 	else
-		LUA_INCLUDE="${LUA_INCLUDE} -I${LUA_DIR}/lua5.1"
+		LUA_INCLUDE="${LUA_INCLUDE} -I${LUA_DIR}/lua51"
 	fi
 fi
 
