diff -urN asterisk-10.10.1.org/menuselect/Makefile asterisk-10.10.1/menuselect/Makefile
--- asterisk-10.10.1.org/menuselect/Makefile	2012-05-09 21:57:43.000000000 +0200
+++ asterisk-10.10.1/menuselect/Makefile	2012-12-09 16:21:44.578265277 +0100
@@ -23,6 +23,7 @@
 # Basic set of sources and flags/libraries/includes
 OBJS:=menuselect.o strcompat.o
 CFLAGS+=-g -D_GNU_SOURCE -Wall
+MXML_LIB:=$(shell pkg-config --libs mxml)
 
 ifeq ($(MENUSELECT_DEBUG),yes)
   CFLAGS += -DMENUSELECT_DEBUG
@@ -44,19 +45,20 @@
 
 ifdef GTK2_LIB
   G_OBJS += menuselect_gtk.o
-  G_LIBS += $(GTK2_LIB)
+  G_LIBS += $(MXML_LIB) $(GTK2_LIB)
   G_INCLUDE += $(GTK2_INCLUDE)
   ALL_TGTS += gmenuselect
 endif
 
 ifdef NEWT_LIB
   N_OBJS += menuselect_newt.o
-  N_LIBS += $(NEWT_LIB)
+  N_LIBS += $(MXML_LIB) $(NEWT_LIB)
   N_INCLUDE += $(NEWT_INCLUDE)
   ALL_TGTS += nmenuselect
 endif
 
 M_OBJS += menuselect_stub.o
+M_LIBS += $(MXML_LIB)
 ALL_TGTS += menuselect
 
 all: $(ALL_TGTS)
@@ -66,7 +68,7 @@
 makeopts autoconfig.h: autoconfig.h.in makeopts.in
 	@./configure $(CONFIGURE_SILENT)
 
-$(ALL_TGTS): mxml/libmxml.a
+$(ALL_TGTS):
 
 ifdef C_OBJS
 menuselect_curses.o: CFLAGS+=$(C_INCLUDE)
diff -urN asterisk-10.10.1.org/menuselect/menuselect.c asterisk-10.10.1/menuselect/menuselect.c
--- asterisk-10.10.1.org/menuselect/menuselect.c	2011-11-09 23:22:13.000000000 +0100
+++ asterisk-10.10.1/menuselect/menuselect.c	2012-12-09 16:22:51.341349575 +0100
@@ -32,7 +32,7 @@
 #include <getopt.h>
 
 #include "autoconfig.h"
-#include "mxml/mxml.h"
+#include "mxml.h"
 #include "linkedlists.h"
 #include "menuselect.h"
 
