diff -dur -x '*~' -x '*.orig' -x '*.rej' asterisk-14.5.0.orig/main/Makefile asterisk-14.5.0/main/Makefile
--- asterisk-14.5.0.orig/main/Makefile	2017-05-30 19:50:46.000000000 +0200
+++ asterisk-14.5.0/main/Makefile	2017-06-09 14:43:16.000000000 +0200
@@ -47,7 +47,7 @@
 AST_LIBS+=$(RT_LIB)
 AST_LIBS+=$(SYSTEMD_LIB)
 
-ifneq ($(findstring $(OSARCH), linux-gnu uclinux linux-uclibc linux-musl kfreebsd-gnu),)
+ifneq ($(findstring $(OSARCH), linux-gnu linux-gnux32 uclinux linux-uclibc linux-musl kfreebsd-gnu),)
   AST_LIBS+=-ldl
   ifneq (x$(CAP_LIB),x)
     AST_LIBS+=$(CAP_LIB)
diff -dur -x '*~' -x '*.orig' -x '*.rej' asterisk-14.5.0.orig/Makefile asterisk-14.5.0/Makefile
--- asterisk-14.5.0.orig/Makefile	2017-05-30 19:50:46.000000000 +0200
+++ asterisk-14.5.0/Makefile	2017-06-09 14:42:55.000000000 +0200
@@ -182,6 +182,11 @@
 # Create OPTIONS variable, but probably we can assign directly to ASTCFLAGS
 OPTIONS=
 
+ifeq ($(OSARCH),linux-gnux32)
+  # flag to tell 'ldconfig' to only process specified directories
+  LDCONFIG_FLAGS=-n
+endif
+
 ifeq ($(findstring -save-temps,$(_ASTCFLAGS) $(ASTCFLAGS)),)
   ifeq ($(findstring -pipe,$(_ASTCFLAGS) $(ASTCFLAGS)),)
     _ASTCFLAGS+=-pipe
